src/components/base/Text/index.tsx

```diff
@@ -53,11 +64,47 @@ const Text: FC<Props> = ({
 
   switch (tag) {
     case 'span':
-      return <span className={className}>{children}</span>;
+      return (
+        <span
+          className={clsx(baseStyles, recipeClassName)}
+          style={assignInlineVars(themeVars, {
+            fontSize: {
+              mobile: vars.fontSize[fontSize.mobile],
+              desktop: vars.fontSize[fontSize.desktop],
+            },
```

Comment:
よくわかってないんだけど、フォントサイズを動的に指定できるようにはなったけど、varsに定義されていないフォントサイズが指定されたら駄目ってこと？

Reply:
ですね。
動的に指定はできるけど、定義されたフォントサイズ内で指定してね🫶というルールみたいなもんっすね。
俺以外の人がコード書くことがあった時に便利かなと

Reply:
これ定義されてないフォントサイズを指定したらどういう挙動をするんですか？
あとコンポーネントを使う側から見ると、そのルールが暗黙的に見えるのがイマイチなのかなーと思うんですよねえ

Reply:
ts使ってる上、定義されてないフォントサイズ指定したらエラーになるので、実行前に検知できる感じっすね
暗黙的に見えるかもしれないけど、使うときには↑の挙動になるんで、問題ないのかな〜という感じはしてる

Reply:
まあ良いのかなあ。裏側で変数定義してるならわざわざ自由な値を渡せる設計にするメリットもあまり無いような気はしたけど

Reply:
若干認識がズレてそうなので補足すると、Dynamics使うと自由な値を渡せるというメリットもあるけど、TS使う上ではanyのような機能でもあるので、それをコンポーネント内で制御してる感じですかね

Reply:
裏側で変数定義できている ≒ 動的にするほど複雑度が高くない、ということかなと思っていて、それならもう入り口を静的にしたほうが良いんじゃないか？と思っている感じですね

Reply:
例えば1〜100まで値として渡せるという話なら入り口を動的にして中でバリデーションして落とす、でも良いと思うんだけどたかだか10種類程度だしなあと

Reply:
なるほどな、理解した🙏
ただフォントサイズの組み合わせを洗い出すのが辛いんよな。。。

Reply:
https://github.com/uyupun/official/pull/312/files#diff-426db9988125a0e9f49d136234eab3eb30477f8944feafdc42e365bac83721cdR44-R56 ここで定義されてるのが全てってことではないのか

Reply:
後は命名どうするか考えるのも辛い気持ち🫠

Reply:
そこで定義はされてるけど、例えばレスポンシブ考慮した時にスマホでは10pxでPCでは12pxのところもあれば、スマホ、PCどちらも10pxのところがあったり。。。みたいな組み合わせを洗い出すのが辛いって感じっすね

Reply:
上記のような組み合わせを洗い出したとしても、命名どうしよ〜みたいな感じでツラミなんよな。。。

Reply:
あー、いや、そういうことではなくて、モバイルとPCそれぞれで変数定義してそれを渡すようにしたら良いんじゃないかなーみたいな

Reply:
一応↓みたいに渡すようにはしてるけど、こういうことではないってこと？
mobileとdesktopは共通のFontSize（添付してくれたフォントサイズを定義してる箇所の型）使ってるので共通ではあるけど。。。
理解が追いついてない。。。すまぬ。。。

```typescript
<Text fontSize={{ mobile: 16, desktop: 18 }}>
  text
</Text>
```

Reply:
その16とか18が変数として定義してあって、それをわたすかんじかな。動くか分からんけど↓みたいな。

```
<Text fontSize={{ mobile: fontSize.16, desktop: fontSize.17 }}>
  text
</Text>
```

それでコンポーネント側はfontSize.mobileとfontSize.desktopで渡された数値がFontSize型じゃなかったら落とすイメージしてた

Reply:
なるほど、理解した。
コンポーネント側は変数を受け取るので、↓みたいになるということか🤔

```suggestion
            fontSize: {
              mobile: fontSize.mobile,
              desktop: fontSize.desktop,
            },
```

ただ、フロント界隈だとあんまり変数を渡すことはなくて（都度importしてこないといけないからかな🤔）、型に沿って値を渡すことが多いんすよね。（1つ前のコメントのやつみたいに）

一応Storybook上でも型が見れるのと、VSCodeなどのエディタで補完が効くので変数使わなくて良いかなというのが個人的な意見かな🤔
<img width="920" alt="スクリーンショット 2023-10-18 10 27 46" src="https://github.com/uyupun/official/assets/30039352/2cfa4b5a-b3fb-470d-86d3-e95a85fc902b">

---

このあたり意見が割れそうなので、以前のようにパターン割り出して`s`/`m`/`l`みたいにした方が良かったりする？
時間かかるからと避けて別パターン取ったけど、Figmaの整理もできて一石二鳥ではあるかなと思い始めてきました💦

Reply:
なるほどなー、仮にs/m/lみたいな分け方した場合、今回のケースだとどういう分け方になるんかなーというのが気になる

Reply:
Dynamics排除する方向に舵切ります🚢
パターン洗い出して、コメントに書きますね〜

Reply:
というか、かずきちのコメントで思ったけど、モバイルとPCの組み合わせでsとかmとかじゃなくて、個別にsとかmとか設定するようにすれば、組み合わせを洗い出さないといけないみたいなことはなくなるのか

Reply:
そうっすね！

Reply:
洗い出したけど、くっそ少なかった🥺🥺（下記以外のサイズはルビだったりタイトルだったりラベルだったりリンクだったり...）
モバイル: 14
PC: 14 / 16

なのでモバイルはm(medium)オンリーでPCがs(small)とｍ(medium)の2つって感じかな
（なんでめっちゃ多いと勘違いしていたんだろうか。。。）

Reply:
ワロタ、まあ今後増える可能性はあるけどいうてそこまでは増えないか。

Reply:
`src/styles/themes.css.ts` の `fontSize` には10種類ぐらいあると思うけど、この全種類に対応させる必要はないのか。

Reply:
全種類に対応する必要はないっすね
タイトルは別でコンポーネントあったりとかやし

Reply:
これ別PRで対応しますね🙏
不要なコミット含めちゃうので

Reply:
なるほど、クローズおねがいします